DTML.EduBot/Dialogs/RootDialog.cs

```diff
@@ -3,76 +3,265 @@
+                    context.Wait(this.UserNameReceivedAsync);
+                }
+                else
+                {
+                    // detected user language
+
+                    CultureInfo[] allCultures =CultureInfo.GetCultures(CultureTypes.AllCultures);
+                    CultureInfo detectedCulture =
+                       allCultures
+                            .Where(cultureInfo => (detectedLanguageIsoCode.Contains(cultureInfo.TwoLetterISOLanguageName)))
+                            .FirstOrDefault();
+
+                    string detectedLanguageName = CultureInfo.GetCultureInfo(MessageTranslator.DEFAULT_LANGUAGE).DisplayName;
+
+                    if (detectedCulture != null)
+                    {
+                        detectedLanguageName = detectedCulture.DisplayName;
+                    }
+
+                    string translatedSwitchQuestion = await MessageTranslator.TranslateTextAsync($"Do you want to switch to {detectedLanguageName}", detectedLanguageIsoCode);
+
+                    string translatedDontUnderstand = await MessageTranslator.TranslateTextAsync("I am sorry but I didn't understand that. I need you to select one of the options below", detectedLanguageIsoCode);
+
+                    PromptDialog.Choice(
+                        context,
+                        this.AfterChoosingLanguageSwitch,
+                        await this.TranslatedChoices(YesNoChoices, detectedLanguageIsoCode),
+                        translatedSwitchQuestion,
+                        translatedDontUnderstand,
+                        attempts: MAX_ATTEMPT
+                    );
+                }
+            }
+        }
+
+        private Task UserNameReceivedAsync(IDialogContext context, IAwaitable<IMessageActivity> result)
+        {
             PromptDialog.Choice(
                 context,
-                this.AfterChoiceSelected,
-                Choices,
-                "Hello Dear " + friendlyUserName + ",\n What would you like to do.",
+                this.AfterDialogChoiceSelectedAsync,
+                DialogChoices,
+                $"Hello Dear {result.GetAwaiter().GetResult().Text},\n What would you like to do.",
                 "I am sorry but I didn't understand that. I need you to select one of the options below",
-                attempts: Choices.Count());
+                attempts: MAX_ATTEMPT);
 
             return Task.CompletedTask;
         }
 
-        private async Task AfterChoiceSelected(IDialogContext context, IAwaitable<string> result)
+        private async Task UserNameReceivedInNativeLanguageAsync(IDialogContext context, IAwaitable<IMessageActivity> result)
+        {
+            string nativeLanguageIsoCode = MessageTranslator.DEFAULT_LANGUAGE;
+            using (var scope = WebApiApplication.FindContainer().BeginLifetimeScope())
+            {
+                UserData.UserData userData = scope.Resolve<IUserDataRepository>().GetUserData(context.Activity.From.Id);
```

Comment:
Just general comment in IoC using Resolve is a bad practice, we should move either to controller or property injection